examples/compositor/main.c

```diff
@@ -39,22 +58,13 @@ void handle_output_frame(struct output_state *output, struct timespec *ts) {
 	wlr_output_make_current(wlr_output);
 	wlr_renderer_begin(sample->renderer, wlr_output);
 
-	struct wl_resource *_res;
-	float matrix[16];
-	wl_list_for_each(_res, &sample->compositor.surfaces, link) {
-		struct wlr_surface *surface = wl_resource_get_user_data(_res);
-		wlr_surface_flush_damage(surface);
-		if (surface->texture->valid) {
-			wlr_surface_get_matrix(surface, &matrix,
-				&wlr_output->transform_matrix, 200, 200);
-			wlr_render_with_matrix(sample->renderer, surface->texture, &matrix);
-
-			struct wlr_frame_callback *cb, *cnext;
-			wl_list_for_each_safe(cb, cnext, &surface->frame_callback_list, link) {
-				wl_callback_send_done(cb->resource, timespec_to_msec(ts));
-				wl_resource_destroy(cb->resource);
-			}
-		}
+	struct wlr_wl_shell_surface *wl_shell_surface;
+	wl_list_for_each(wl_shell_surface, &sample->wl_shell.surfaces, link) {
```

Comment:
wl_shell's resource list should be a list of wl_shell_surfaces, not a list of wl_surfaces. The link conflict should not be a problem.